### ISSUE TITLE
snapshot scheduled task regression re 3.9.1-9 API change

### ISSUE THREAD

**Issue by phillxnet** [rockstor/rockstor-core#1809]
As of Rockstor version 3.9.1-9 there is a regression in scheduled snapshot tasks. This is due to a change in share API from name to id which needs to be accompanied by a pending change in task definition. Previously we used the the 'share' field within json_meta in smart_manager_taskdefinition to store ie: "share": "luks-share" as the name of the share to be snapshot-ed however given the API change we can no longer use this same element to effect API creation.

A parallel scheduled pool scrub regression was indicated in issue:
#1759 "Scheduled scrub throws error"
and consequently fixed in pr:
#1760 "fix scheduled scrub regression re pool api changes. Fixes #1759"

The fix in that instance was to move existing use of json_meta .pool to reference id and add json_meta.pool_name (during task creation) so that scrub tasks held pool name info for display purposes. This is not an ideal fix in light of the pending pool / share re-name capability but did serve to perpetuate pool name display in task related tables.

It is suggested that we enable a similar quick, if less than ideal, fix in this scenario and look to enabling a fix that honours the pending share name change feature when that feature is introduced.

Current consequence is that all existing and freshly created snapshot scheduled tasks will fail to execute with the indication of such being an "error" task status upon scheduled task execution time.

Manual snapshot creation is not affected.

**Comment by phillxnet** [comment]
@schakrava I propose we prioritise this issue sufficiently to enable it's fix to be included prior to the next stable release.

**Comment by phillxnet** [comment]
Adding example of email notification re cron failure as a result of this regression:

Subject: /opt/rockstor-dev/bin/st-snapshot 12 \*-*-*-*-*-*

Contents:
```
Traceback (most recent call last):
  File "/usr/lib64/python2.7/logging/handlers.py", line 76, in emit
    if self.shouldRollover(record):
  File "/usr/lib64/python2.7/logging/handlers.py", line 154, in shouldRollover
    msg = "%s\n" % self.format(record)
  File "/usr/lib64/python2.7/logging/__init__.py", line 724, in format
    return fmt.format(record)
  File "/usr/lib64/python2.7/logging/__init__.py", line 464, in format
    record.message = record.getMessage()
  File "/usr/lib64/python2.7/logging/__init__.py", line 324, in getMessage
    msg = str(self.msg)
TypeError: __str__ returned non-string (type list)
Logged from file snapshot.py, line 118
```

as per issue #1798 which was closed as a duplicate of #1759 except in this issue relating to snapshot of share not scrub of pool, but the same mechanism is used for both.

**Comment by phillxnet** [comment]
I am currently working on this issue.
